src/colvarbias_meta.cpp

```diff
@@ -561,36 +563,40 @@ int colvarbias_meta::update_bias()
     }
 
     if (well_tempered) {
+
       cvm::real hills_energy_sum_here = 0.0;
-      if (use_grids) {
-        std::vector<int> curr_bin = hills_energy->get_colvars_index();
-        const bool index_ok = hills_energy->index_ok(curr_bin);
-        if (index_ok) {
-          // TODO: Should I sum the energies from other replicas?
-          hills_energy_sum_here = hills_energy->value(curr_bin);
-        } else {
-          if (!keep_hills) {
-            // TODO: Should I sum the off-grid hills from other replicas?
-            calc_hills(hills_off_grid.begin(),
-                       hills_off_grid.end(),
-                       hills_energy_sum_here,
-                       &colvar_values);
+
+      for (size_t ir = 0; ir < replicas.size(); ir++) {
+        if (use_grids) {
+          std::vector<int> curr_bin = hills_energy->get_colvars_index();
+          const bool index_ok = hills_energy->index_ok(curr_bin);
+
+          if (index_ok) {
+            hills_energy_sum_here += replicas[ir]->hills_energy->value(curr_bin);
           } else {
-            // TODO: Is it better to compute the energy from all historic hills
-            //       when keepHills is on?
-            calc_hills(hills.begin(),
-                       hills.end(),
-                       hills_energy_sum_here,
-                       &colvar_values);
+            if (!keep_hills) {
+              calc_hills(replicas[ir]->hills_off_grid.begin(), replicas[ir]->hills_off_grid.end(),
+                         hills_energy_sum_here, &colvar_values);
+            } else {
+              // TODO: Is it better to compute the energy from all historic hills
+              //       when keepHills is on?
```

Comment:
@HanatoK I didn't understand this comment before, and forgot to ask you about it